src/devices/Ssd1351/Ssd1351.cs

```diff
@@ -0,0 +1,325 @@
+        private ColorDepth _colorDepth;
+        private ColorSequence _colorSequence;
+
+        private const byte SCREEN_WIDTH_PX = 128;
+        private const byte SCREEN_HEIGHT_PX = 128;
+        private const int DefaultSPIBufferSize = 0x1000;
+
+        /// <summary>
+        /// Initializes new instance of Ssd1351 device that will communicate using SPI bus.
+        /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+        /// light emitting diode dot-matrix graphic display system. 
+        /// </summary>
+        /// <param name="spiDevice">The SPI device used for communication.</param>
+        /// <param name="gpioController">The GPIO controller used for communication.</param>
+        /// <param name="dataCommandPin">The id of the GPIO pin used to control the DC line (data/command).</param>
+        /// <param name="resetPin">The id of the GPIO pin used to control the /RESET line (data/command).</param>
+        /// <param name="spiBufferSize">The size of the SPI buffer. If data larger than the buffer is sent then it is split up into multiple transmissions. The default value is 4096.</param>
+        public Ssd1351(SpiDevice spiDevice, IGpioController gpioController, int dataCommandPin, int resetPin, int spiBufferSize = DefaultSPIBufferSize)
+        {
+            if (!InRange((uint)spiBufferSize, 0x1000, 0x10000))
+            {
+                throw new ArgumentException($"SPI Buffer Size must be between 4096 and 65536.", nameof(spiBufferSize));
+            }
+
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+
+            _dcPinId = dataCommandPin;
+            _resetPinId = resetPin;
+
+            _gpioDevice = gpioController;
+
+            _gpioDevice.OpenPin(_dcPinId, PinMode.Output);
+            _gpioDevice.OpenPin(_resetPinId, PinMode.Output);
+
+            _spiBufferSize = spiBufferSize;
+
+            SendCommand(new SetSegmentReMapColorDepth());
+        }
+
+        /// <summary>
+        /// Initializes new instance of Ssd1351 device that will communicate using SPI bus.
+        /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+        /// light emitting diode dot-matrix graphic display system. 
+        /// </summary>
+        /// <param name="spiDevice">The SPI device used for communication.</param>
+        /// <param name="dataCommandPin">The id of the GPIO pin used to control the DC line (data/command).</param>
+        /// <param name="resetPin">The id of the GPIO pin used to control the /RESET line (data/command).</param>
+        /// <param name="spiBufferSize">The size of the SPI buffer. If data larger than the buffer is sent then it is split up into multiple transmissions. The default value is 4096.</param>
+        public Ssd1351(SpiDevice spiDevice, int dataCommandPin, int resetPin, int spiBufferSize = DefaultSPIBufferSize) : this(spiDevice, new GpioController(), dataCommandPin, resetPin, spiBufferSize)
```

Comment:
Instead of having two constructors, just keep one and move the IGpioController parameter to the end of the parameter list in the other constructor and default it to null. Inside the constructor, when this param is null then initialize a new GpioController and set _disposeGpioController to true.

Reply:
@joperezr 
Uggh. I always like to have the controller close to the pins it controls in the parameter list... Anyway I have made the change you suggest in the lates PR update